### PR TITLE
ping: Check if stm32flash exist before bootloader and flash call

### LINKED ISSUE #407: Check if stm32flash exist before executing it and before calling bootloader message

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #407 

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by patrickelectric** [comment]
- [x] Tested on Linux  
- [ ] Tested on Windows 
- [ ] Tested on Mac

**Comment by jaxxzer** [comment]
Have you run into a case where it does not exist? (outside of dev)

File: src/sensor/ping.cpp [hunk 1]
```diff
@@ -447,14 +468,12 @@ void Ping::flash(const QString& portLocation, const QString& firmwareFile, int b
...
 
-#ifdef Q_OS_OSX
-    // macdeployqt file do not put stm32flash binary in the same folder of pingviewer
-    static QString binPath = '"' + QCoreApplication::applicationDirPath() + "/../..";
-#else
-    static QString binPath = '"' + QCoreApplication::applicationDirPath();
-#endif
-    static QString cmd = binPath + QStringLiteral("/stm32flash\" -w \"%0\" %1 -g 0x0 -b %2 %3").arg(
-                             firmwareFileInfo.absoluteFilePath(), verify ? "-v" : "", QString::number(baud), portLocation
+    static QString cmd = QStringLiteral("\"%4\" -w \"%0\" %1 -g 0x0 -b %2 %3").arg(
```
Comment (jaxxzer) [inline src/sensor/ping.cpp]: Can we put the args in ascending order please?

**Comment by patrickelectric** [comment]
Nops, the idea is to help us to debug this issue in development, and to create a debug message for us and for the user for future problems.

**Pull request merged**